tests/state/test_rate_limit.py

```diff
@@ -121,11 +125,63 @@ def test_rate_limit_container(self) -> None:
         assert rate_limit_container.get_stats("foo") == rate_limit_stats
         assert rate_limit_container.get_stats("bar") is None
 
-        assert rate_limit_container.to_dict() == {"foo_rate": 0.5, "foo_concurrent": 2}
+        assert dict(rate_limit_container.to_dict()) == {
+            "foo_rate": 0.5,
+            "foo_concurrent": 2,
+        }
 
     def test_bypass_rate_limit(self) -> None:
         rate_limit_params = RateLimitParameters("foo", "bar", None, None)
         state.set_config("bypass_rate_limit", 1)
 
         with rate_limit(rate_limit_params) as stats:
             assert stats is None
+
+    def test_rate_limit_exceptions(self) -> None:
+        params = RateLimitParameters("foo", "bar", None, 5)
+        bucket = "{}{}".format(state.ratelimit_prefix, params.bucket)
```

Comment:
Please have a function in the rate_limit.py module that generates the bucket name and that is used both in the test and in the production code. Otherwise we would have to change two places for no reason if we change it.